cmd/doracled/cmd/gen_oracle_key.go

```diff
@@ -0,0 +1,112 @@
+	Long: `Generate a new pair of oracle key and its remote report. 
+If the sealed oracle private key exist already, this command will replace the existing one.
+So please be cautious in using this command.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// get config
+		conf, err := config.ReadConfigTOML(getConfigPath())
+		if err != nil {
+			return fmt.Errorf("failed to read config from file: %w", err)
+		}
+
+		// get proper path for oracle key
+		oraclePrivKeyPath, oraclePubKeyPath := getKeyPaths(conf.Enclave.Enable)
+
+		// If there is the existing oracle key, double-check for generating a new oracle key
+		if tos.FileExists(oraclePrivKeyPath) {
+			buf := bufio.NewReader(os.Stdin)
+			ok, err := input.GetConfirmation("This can replace the existing oracle-key.sealed file.\nAre you sure to make a new oracle key?", buf, os.Stderr)
+
+			if err != nil || !ok {
+				log.Printf("Oracle key generation is canceled.")
+				return err
+			}
+		}
+
+		// generate a new oracle key
+		oraclePrivKey, err := secp256k1.NewPrivKey()
+		if err != nil {
+			log.Errorf("failed to generate oracle key: %v", err)
+			return err
+		}
+
+		// seal and store oracle private key
+		if err := sgx.SealToFile(oraclePrivKey.Serialize(), oraclePrivKeyPath, conf.Enclave.Enable); err != nil {
+			log.Errorf("failed to write %s: %v", oraclePrivKeyPath, err)
+			return err
+		}
+
+		// generate oracle key remote report
+		oraclePubKey := oraclePrivKey.PubKey().SerializeCompressed()
+		oracleKeyRemoteReport, err := sgx.GenerateRemoteReport(oraclePubKey[:], conf.Enclave.Enable)
+		if err != nil {
+			log.Errorf("failed to generate remote report of oracle key: %v", err)
+			return err
+		}
+
+		// store oracle pub key and its remote report to a file
+		if err = storeOraclePubKey(oraclePubKey, oracleKeyRemoteReport, oraclePubKeyPath); err != nil {
+			log.Errorf("failed to save oracle pub key and its remote report: %v", err)
+			return err
+		}
+
+		return nil
+	},
+}
+
+// storeOraclePubKey stores base64-encoded oracle public key and its remote report
+func storeOraclePubKey(oraclePubKey, oracleKeyRemoteReport []byte, path string) error {
+	oraclePubKeyData := OraclePubKeyInfo{
+		PublicKey:    base64.StdEncoding.EncodeToString(oraclePubKey),
+		RemoteReport: base64.StdEncoding.EncodeToString(oracleKeyRemoteReport),
+	}
+
+	oraclePubKeyFile, err := json.Marshal(oraclePubKeyData)
+	if err != nil {
+		return fmt.Errorf("failed to marshal oracle pub key data: %w", err)
+	}
+
+	err = ioutil.WriteFile(path, oraclePubKeyFile, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write oracle pub key file: %w", err)
+	}
+
+	return nil
+}
+
+func getKeyPaths(enclaveEnabled bool) (string, string) {
+	if enclaveEnabled {
+		return types.DefaultOraclePrivKeyFilePath, types.DefaultOraclePubKeyFilePath
+	} else {
+		return filepath.Join(homeDir, types.DefaultOraclePrivKeyName), filepath.Join(homeDir, types.DefaultOraclePubKeyName)
+	}
+}
```

Comment:
Is there a reason you separated this path?
I think it can be used in the same way in the enclave environment as below.
```go
func getKeyPaths() (string, string) {
    return filepath.Join(homeDir, types.DefaultOraclePrivKeyName), filepath.Join(homeDir, types.DefaultOraclePubKeyName)
}
```

Because it sets the HOME environment variable of encalve.json.

Reply:
You're right. Regardless of the environments, the `homeDir` is set properly. So we don't need to separate two cases.
I fixed it [5341f29](https://github.com/medibloc/panacea-doracle/pull/4/commits/5341f2945bd0d1ca50e44c1871636c92764df5f6)